src/transforms/filter.jl

```diff
@@ -0,0 +1,44 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter{F} <: Transform
+  pred::F
+end
+
+isrevertible(::Type{Filter}) = true
+
+function apply(transform::Filter, input_table)
+  # Converting to Tables.rowtable to allow length function and indexing 
+  table = Tables.rows(input_table)
+
+  l = length(table)
```

Comment:
Are you certain that the length function is part of the row table API? I thought that a row table was only iterable without a known length.

Reply:
I have tested it and it does work, although I don't know whether it is intended to be used. Do you think I should replace it with something else?